src/views/tutorial-view/index.tsx

```diff
@@ -33,64 +28,81 @@ import TutorialsSidebar, {
 import TabProvider from 'components/tabs/provider'
 import TutorialMeta from 'components/tutorial-meta'
 import VideoEmbed from 'components/video-embed'
-import InstruqtProvider from 'contexts/instruqt-lab'
-import { useOptInAnalyticsTracking } from 'hooks/use-opt-in-analytics-tracking'
 import { getLearnRedirectPath } from 'components/opt-in-out/helpers/get-learn-redirect-path'
 
 // Local imports
+import {
+	CollectionContext,
+	LayoutContentWrapperProps,
+	TutorialData,
+	TutorialSidebarSidecarProps,
+	TutorialViewProps,
+} from './types'
 import MDX_COMPONENTS from './utils/mdx-components'
 import { formatTutorialToMenuItem, generateCanonicalUrl } from './utils'
+import getVideoUrl from './utils/get-video-url'
+import { getCanonicalCollectionSlug } from './utils/get-canonical-collection-slug'
 import {
 	FeaturedInCollections,
 	NextPrevious,
 	getNextPrevious,
 } from './components'
-import getVideoUrl from './utils/get-video-url'
-import { getCanonicalCollectionSlug } from './utils/get-canonical-collection-slug'
 import s from './tutorial-view.module.css'
 
-export interface TutorialViewProps {
-	layoutProps: TutorialSidebarSidecarProps
-	product: LearnProductData
-	tutorial: TutorialData
-}
+/**
+ * The purpose of this wrapper component is to make it possible to invoke the
+ * `useMobileMenu` hook. It cannot be invoked in `TutorialView`. This is because
+ * it requires being invoked within a `MobileMenuProvider`.
+ *
+ * `MobileMenuProvider` is rendered by `CoreDevDotLayout`, which is indirectly
+ * rendered by `SidebarSidecarLayout`. This means the `useMobileMenu` hook can
+ * only be invoked here by a component rendered within `SidebarSidecarLayout`.
+ */
+const LayoutContentWrapper = ({
+	children,
+	collectionCtx,
+	product,
+	setCollectionViewSidebarSections,
+}: LayoutContentWrapperProps) => {
+	const { mobileMenuIsOpen } = useMobileMenu()
+	const haveLoadedData = useRef(false)
```

Comment:
💭 Generally booleans are named using `has` or `is` e.g.: `hasLoadedData`, `isDataLoaded`

I like that you chose to use a ref to avoid any unnecessary state updates 👍

Reply:
Oop! Yes. Didn't have my tenses right when I typed this out, thank you! 😁 Updated in e733359617bc58940421f268e4b7a2cbb307eec6!